docs/architecture/adr-015-namespace-id-size.md

```diff
@@ -0,0 +1,64 @@
+# ADR 15: Namespace ID Size
+
+## Status
+
+Proposed
+
+## Changelog
+
+- 2023/2/17: initial draft
+
+## Context
+
+Namespace ID is currently an 8 byte slice. 8 bytes provides a maximum of 2^64 possible namespace IDs. In practice some namespace IDs are reserved for protocol use so the number of namespace IDs available to users is 2^56 - 2. Modifying the size of a namespace ID post-launch is a breaking change and has implications for the NMT, share encoding, etc. so we'd like to carefully consider the size of the namespace ID pre-launch.
+
+Desirandum:
+
+1. Ability to randomly generate a new namespace that hasn't been used before
+
+// what other desirandum do we have?
+
+## Notes
+
+- The namespace ID must provide at least 72 bits of randomness ([Eager](https://eager.io/blog/how-long-does-an-id-need-to-be/)) to satisfy desirandum 1. Since an 8 byte namespace ID can only provide 64 bits of randomness, it fail to meet this desirandum.
```

Comment:
Can you please provide a link to the `Desirandum` definition? or add a short explanation in the text on what `satisfy desirandum 1` means.

Reply:
I spelled it incorrectly but I meant to use: https://www.dictionary.com/browse/desideratum
I can replace with something more explicit like "Desirable criteria"